pkg/transport/generic_transport_controller.go

```diff
@@ -975,7 +1056,7 @@ func setLabel(object metav1.Object, key string, value any) {
 }
 
 // cleanObject is a function to clean object before adding it to a wrapped object. these fields shouldn't be propagated to WEC.
-func cleanObject(object *unstructured.Unstructured) *unstructured.Unstructured {
+func (ctc *customTransformCollection) cleanObject(ctx context.Context, groupResource metav1.GroupResource, object *unstructured.Unstructured, bindingName string) *unstructured.Unstructured {
```

Comment:
I think we should differentiate between cleaning objects and transforming them.
The transformation should occur in a new function, and the order of first cleaning and then transforming should be preserved (as it is below).

Reply:
We have had some discussion of terminology, perhaps not all in your presence. @ezrasilvera was strongly of the opinion that the feature should be called "transformation", and others agreed. We also agreed that we do not currently have a use case for doing more than removing content, and this is a subset of what is called "transformation".

Reply:
The force-push to ecaf86fcc updates the name of this method and the comments on it.

Reply:
BTW, this method should ultimately be in the other file; I left it here for now in order to spare reviewers from more verbose `diff` output than necessary.

Reply:
On further reflection and doc writing, I have realized that this method is a client of the instructions cache. I re-framed it as such in today's pushes.